charts/metallb/templates/speaker.yaml

```diff
@@ -0,0 +1,131 @@
+{{- if .Values.controller.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ template "metallb.fullname" . }}-speaker
+  labels:
+    {{- include "metallb.labels" . | nindent 4 }}
+    app.kubernetes.io/component: speaker
+spec:
+  selector:
+    matchLabels:
+      {{- include "metallb.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: speaker
+  template:
+    metadata:
+      {{- if or .Values.prometheus.scrapeAnnotations .Values.speaker.podAnnotations }}
+      annotations:
+        {{- if .Values.prometheus.scrapeAnnotations }}
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "7472"
+        {{- end }}
+        {{- with .Values.podAnnotations }}
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+      {{- end }}
+      labels:
+        {{- include "metallb.selectorLabels" . | nindent 8 }}
+        app.kubernetes.io/component: speaker
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ template "metallb.speaker.serviceAccountName" . }}
+      terminationGracePeriodSeconds: 0
+      hostNetwork: true
+      containers:
+      - name: speaker
+        image: {{ .Values.speaker.image.repository }}:{{ .Values.speaker.image.tag | default .Chart.AppVersion }}
+        imagePullPolicy: {{ .Values.speaker.image.pullPolicy }}
+        args:
+        - --port=7472
+        - --config={{ template "metallb.configMapName" . }}
+        env:
+        - name: METALLB_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: METALLB_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: METALLB_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        - name: METALLB_ML_BIND_ADDR
```

Comment:
We need a "enable memberlist" boolean, set to true by default, that if set to false remove all the METALLB_ML*
We've had pretty good results with memberlist but BGP only user don't need it, and it's nice to have it for debug

Reply:
I'll add that, thanks.  Will memberlist just not be enabled if the `METALLB_ML*` variables aren't present?

The switch will be in `.Values.metallb.speaker.memberlist.enabled` (boolean).

Reply:
> Will memberlist just not be enabled if the METALLB_ML* variables aren't present?

yes
https://github.com/metallb/metallb/blob/main/internal/speakerlist/speakerlist.go#L54

Reply:
Added switch to disable these vars:

speaker - METALLB_ML_BIND_ADDR METALLB_ML_LABELS METALLB_ML_BIND_PORT METALLB_ML_SECRET_KEY
controller - METALLB_ML_SECRET_NAME